src/turtle.rs

```diff
@@ -1156,4 +1158,22 @@ mod tests {
         assert_eq!(turtle.position()[1].round(), 100.0);
         assert_eq!(turtle.heading(), 51.0);
     }
+
+    #[test]
+    fn turn_towards() {        
```

Comment:
Thanks for updating this! I think you need to move lines 1168-1170 into the inner loop. Otherwise you are rotating to original_angle only once, then essentially rotating in a circle without actually going back to the original angle. The goal is to test that you can rotate from any angle to any angle. That means we need to go back to the original angle every time in order to make an exhaustive test. 

If you make that change, you'll find that the test actually doesn't pass anymore. In fact, I made a little demo application which reveals that a bug still remains in the implementation of `turn_towards`.

Try running this by overwriting the contents of an example like `examples/circle.rs` and then running it with `cargo run --example circle` (don't commit that change, this is just to help you debug) Click to advance to each rotation. This should be a really helpful tool for you to debug what is going on. Feel free to modify it as you see fit in order to help you figure out the problem.

```rust
extern crate turtle;
use turtle::Turtle;

use std::f64::consts::PI;

fn main() {
    let mut turtle = Turtle::new();

    // Turn from each cardinal direction to each cardinal direction
    for n in 0..16 as u32 {
        let original_angle = 2.*PI * n as f64 / 16.0;
        for i in 0..16 as u32 {
            println!("{:?}", (n, i));
            turtle.wait_for_click();

            turtle.turn_towards([original_angle.cos(), original_angle.sin()]);
            //assert_eq!(turtle.heading().ceil(), original_angle.to_degrees().ceil());
            turtle.wait(1.0);

            let target_angle = 2.*PI * i as f64 / 16.0;
            turtle.turn_towards([target_angle.cos(), target_angle.sin()]);
            //assert_eq!(turtle.heading().ceil(), target_angle.to_degrees().ceil());
            turtle.forward(100.0);
            turtle.backward(100.0);
        }
        turtle.wait_for_click();
        turtle.clear();
    }
}
```

![peek 2017-12-10 12-36](https://user-images.githubusercontent.com/530939/33807582-f4e5b016-dda6-11e7-9045-38c696edada1.gif)

Notice that this doesn't produce a full circle of 16 lines. There are "gaps" which are the result of a bug in the calculation that still needs to be fixed. Good thing we're writing these tests! If you hadn't volunteered to do that, we wouldn't have found this problem. Thank you so much!